process/block/shardblock.go

```diff
@@ -38,6 +39,9 @@ type shardProcessor struct {
 	txsPoolsCleaner     process.PoolsCleaner
 
 	lowestNonceInSelfNotarizedHeaders uint64
+
+	startOfEpochData      map[string]startOfEpochData
```

Comment:
move all the data gathering to the validator statistics pre processor.

Reply:
fixed